transaction.go

```diff
@@ -0,0 +1,53 @@
+package storm
+
+// Begin starts a new transaction.
+func (n *Node) Begin(writable bool) (*Node, error) {
```

Comment:
> Or: Keep your api as is: db.Begin, Commit (on both db and node) and make a node copy:

It's already done here. Begin returns a new `Node` copy for the transaction. Here is an example from the README:

```go
tx, err := db.Begin(true)

accountA.Amount -= 100
accountB.Amount += 100

err = tx.Save(accountA)
if err != nil {
  tx.Rollback()
}

err = tx.Save(accountB)
if err != nil {
  tx.Rollback()
}

tx.Commit()
```

`tx` is a `*Node` copy, not a `*bolt.Tx`